xclim/sdba/_processing.py

```diff
@@ -74,17 +75,19 @@ def _adapt_freq(
         pth = nbu.vecquantiles(ds.ref, P0_sim, dim).where(dP0 > 0)
 
         # Probabilities and quantiles computed within all dims, but correction along the first one only.
-        if "window" in dim:
-            # P0_sim was computed using the window, but only the original time series is corrected.
-            # Grouper.apply does this step, but if done here it makes the code faster.
-            sim = ds.sim.isel(window=(ds.sim.window.size - 1) // 2)
-        else:
-            sim = ds.sim
-        dim = dim[0]
-
+        sim = ds.sim
         # Get the percentile rank of each value in sim.
-        rank = sim.rank(dim, pct=True)
-
+        if len(dim) > 1:
+            temp_dim = get_temp_dimname(sim.dims, "temp")
+            rank = (
+                sim.stack(**{temp_dim: dim})
+                .rank(temp_dim, pct=True)
+                .unstack(temp_dim)
+                .transpose(*sim.dims)
+                .drop_vars([dim_ for dim_ in dim if dim_ not in sim.coords])
+            )
+        else:
+            rank = sim.rank(dim[0], pct=True)
```

Comment:
Should I incorporate multi-dim ranking in the `sdba.rank` function?

I'm thinking : 
```python
def rank(...
dim : str | list[str]
)
...
dim = dim if isinstance(dim,list) else [dim]
# then use code above with general `pct` 
```

Reply:
I think this is a good idea.